pkg/agent/plugin/workloadattestor/k8s/k8s.go

```diff
@@ -112,14 +111,12 @@ func (p *k8sPlugin) Attest(req *workloadattestor.AttestRequest) (*workloadattest
 			if containerID == containerURL.Host {
 				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("sa:%v", item.Spec.ServiceAccountName)})
 				resp.Selectors = append(resp.Selectors, &common.Selector{Type: selectorType, Value: fmt.Sprintf("ns:%v", item.Metadata.Namespace)})
-				log.Printf("Selectors found: %v", resp.Selectors)
 				return &resp, nil
 			}
 		}
 	}
 
-	log.Print("No selectors found")
-	return &resp, nil
+	return &resp, fmt.Errorf("no selectors found")
```

Comment:
I'm wondering if we have to return an error if no selectors are found when reaching this point. This will be reported as a failure to collect the selector.
Maybe not finding a matching container ID is not an error condition?

Reply:
Hmm... 🤔 

Under what conditions do you think this can occur? I put it like this because I thought maybe it could indicate that the podInfo format had changed or something? If you think this could be a normal condition, we can just remove the message altogether

Reply:
I just wanted to make sure that we were not overlooking this :)
After thinking about this more and looking at the pod list json, looks like in fact at that point a selector should have been found, so the error makes sense.